File: frontend/src/types/lead.ts
import type { DealProduct, PickupDelivery } from '@/types/deal'
import type { StoredAttachment, StoredActivity } from '@/types/attachment'

export type LeadAttachment = StoredAttachment
export type LeadActivityEntry = StoredActivity

export interface LeadProduction {
  nomenclature: string
  dueAt: number | null
  employee: string
}

export interface NewLeadForm {
  firstName: string
  patronymic: string
  phone: string
  trafficSource: string
}

export interface Lead extends NewLeadForm {
  id: string
  leadNumber: number
  columnId: string
  leadComments: string
  failureReason: string
  createdBy: string
  createdAt: number
  updatedAt: number
  pickupDelivery: PickupDelivery
  products: DealProduct[]
  production: LeadProduction
  attachments: LeadAttachment[]
  activities: LeadActivityEntry[]
}

File: backend/internal/model/lead.go
package model

type Lead struct {
	ID             string         `json:"id"`
	LeadNumber     int64          `json:"leadNumber"`
	FirstName      string         `json:"firstName"`
	Patronymic     string         `json:"patronymic"`
	Phone          string         `json:"phone"`
	TrafficSource  string         `json:"trafficSource"`
	ColumnID       string         `json:"columnId"`
	LeadComments   string         `json:"leadComments"`
	FailureReason  string         `json:"failureReason"`
	CreatedBy      string         `json:"createdBy"`
	CreatedAt      int64          `json:"createdAt"`
	UpdatedAt      int64          `json:"updatedAt"`
	PickupDelivery PickupDelivery `json:"pickupDelivery"`
	Products       []DealProduct  `json:"products"`
	Production     DealProduction `json:"production"`
}

type CreateLeadInput struct {
	FirstName     string `json:"firstName"`
	Patronymic    string `json:"patronymic"`
	Phone         string `json:"phone"`
	TrafficSource string `json:"trafficSource"`
	ColumnID      string `json:"columnId"`
	CreatedBy     string `json:"createdBy"`
}
